code_review/dp/code_review_daw_parms.Rmd

```diff
@@ -0,0 +1,280 @@
+
+1. Fit ML Log-Reg model
+2. Extract fixed and random effects
+3. Add fixed effects to random effects to create MB/MF/FSS scores
+
+FSS: The fixed intercept plus the random intercept
+
+MF: The fixed main effect of reward plus the random effect of reward
+
+MB: The fixed interaction effect of reward and transition type plus the random effect of reward and transition type 
+
+
+```{r ML Logistic Regression Model}
+# models
+intmodeltoplot <- glmer(firststagestay ~ 1 + commonraredummy*moneylagdummy + 
+                          (commonraredummy*moneylagdummy|id:visitnum),
+                        data=luna_final,
+                        family="binomial",
+                        glmerControl(optimizer = "bobyqa"))
+summary(intmodeltoplot)
+
+#Marginal effect: Common vs Rare
+cr_me <- ggpredict(intmodeltoplot, terms = "commonraredummy")
+ggplot(cr_me, 
+       aes(x = as.factor(x), 
+           y = predicted)) + 
+  geom_point(size = 3) + 
+  geom_errorbar(aes(ymin = conf.low, 
+                    ymax = conf.high), 
+                width = 0.2) +
+  labs(title = "Main effect of tranistion type",
+    x = "Transition Type",
+    y = "Probability of FSS") +
+   scale_y_continuous(labels = scales::percent) +
+  theme_modern()
+
+#Marginal effect: Reward
+rew_me <- ggpredict(intmodeltoplot, terms = "moneylagdummy")
+ggplot(rew_me, 
+       aes(x = as.factor(x), 
+           y = predicted)) + 
+  geom_point(size = 3) + 
+  geom_errorbar(aes(ymin = conf.low, 
+                    ymax = conf.high), 
+                width = 0.2) +
+  labs(title = "Main effect of reward",
+    x = "Rewarded",
+    y = "Probability of FSS") +
+   scale_y_continuous(labels = scales::percent) +
+  theme_modern()
+
+#Marginal effect: Reward x Transition Type
+cr_rew_me <- ggpredict(intmodeltoplot, terms = c("moneylagdummy", "commonraredummy"))
+ggplot(cr_rew_me, 
+       aes(x = as.factor(x), 
+           y = predicted,
+           group = group,
+           colour = group)) + 
+  geom_point(size = 3) + 
+  geom_errorbar(aes(ymin = conf.low, 
+                    ymax = conf.high), 
+                width = 0.2) +
+  labs(title = "Transition type x reward interaction",
+    x = "Reward",
+    y = "Probability of FSS") +
+   scale_y_continuous(labels = scales::percent) +
+  scale_color_discrete(name = "Transition Type") +
+  theme_modern()
+```
+
+Fun!
+
+
+Extract fixed and random effects. Then, add fixed effects to random effects to create MB/MF/FSS scores
+
+
+```{r Extract Fixed and Random Effects}
+#Extracting random effects
+dawranefint <- ranef(intmodeltoplot)
+head(dawranefint$`id:visitnum`)
+
+#Extracting fixed effects
+dawfixedeffectsint <- summary(intmodeltoplot)$coefficients
+dawfixedeffectsint
+
+#Creating dataframe consisting of MB/MF/Habit parameter.
+dawsubjecterms <- data.frame(
+  idtemp = row.names(dawranefint$`id:visitnum`), 
+  modelbased = dawranefint$`id:visitnum`$`commonraredummy:moneylagdummy` + 
+    dawfixedeffectsint[4,1], 
```

Comment:
use names from `str(dawfixedeffectsint)`  like `dawfixedeffectsint["(Intercept)", "Estimate"]` ?

Reply:
NB. from val&ash. all in log-odds